### ISSUE TITLE
FrameTimeCode should be identic to itself when applying the resulting values to create a new FrameTimeCode

### ISSUE THREAD

**Issue by Wirg** [Breakthrough/PySceneDetect#268]
Hi !

Thank you for PySceneDetect.

I ran into issues where after some serialization the resulting time frame was not the expected one.
For example:
```python
frame_time_code = FrameTimecode(61, fps=14)
seconds = frame_time_code.get_seconds()
time_code = frame_time_code.get_timecode()

...

FrameTimecode(seconds, fps=14).get_frames() == 60 # not 61
FrameTimecode(time_code, fps=14).get_frames() == 60 # not 61
```

I found the corresponding code with flooring (with `int` instead of `round`) :

https://github.com/Breakthrough/PySceneDetect/blob/f6c61dda5ddb2e1bc99471c5520cebe8a9ea7f59/scenedetect/frame_timecode.py#L227-L236

https://github.com/Breakthrough/PySceneDetect/blob/f6c61dda5ddb2e1bc99471c5520cebe8a9ea7f59/scenedetect/frame_timecode.py#L307

I ran into the following code which seems to indicate this is on purpose. What is the reason?

https://github.com/Breakthrough/PySceneDetect/blob/f6c61dda5ddb2e1bc99471c5520cebe8a9ea7f59/tests/test_frame_timecode.py#L114-L116

Have a nice day !

**Comment by Breakthrough** [comment]
Hey @Wirg;

Thanks for looking into this.  It looks like some of the existing test cases are failing after your change (probably just a few off-by-one errors now), so will need to see why that is before merging #269 (thanks for the PR as well).  I'm not sure if this should be taken for v0.5 or not, as this could possibly be a breaking change, but I'd be happy to consider it for v0.6.

I ended up doing a similar workaround in v0.6 for the PyAV backend, so I think your change is correct.  Truthfully it's been quite a few years since I've touched FrameTimecode, so I can't really remember the intent behind these test cases you pointed out:

https://github.com/Breakthrough/PySceneDetect/blob/f6c61dda5ddb2e1bc99471c5520cebe8a9ea7f59/tests/test_frame_timecode.py#L114-L116

I may mark this as a known issue for v0.5, but bring your change in [for v0.6](https://github.com/Breakthrough/PySceneDetect/tree/v0.6).  Thanks for the report!

**Comment by Breakthrough** [comment]
I've accepted your suggested solution for v0.6 (c5ac42b566a4c6d2c0ce64242a0cdf3b59bfe2c9) and it will be included in the next beta release, as well as v0.6.  I'm reluctant to merge this fix into v0.5 as it may constitute a breaking change for some existing applications.

TODO before closing this issue:
 - [X] Add identity unit tests as @Wirg suggested in #269
 - [ ] Update v0.5 changelog marking this as a known issue

**Comment by Wirg** [comment]
Thanks for the fast answer!

I think it's ok for me, I agree with your reasoning. I may try the v0.6 branch then.

**Comment by Breakthrough** [comment]
This is now available in the latest release of v0.6 (v0.6-rc0), feel free to give it a try!
